hnn_core/opt_toy_example/general.py

```diff
@@ -0,0 +1,365 @@
+"""Parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from metrics import _rmse_evoked  # change path***
+
+from skopt import gp_minimize
+from scipy.optimize import fmin_cobyla
+
+
+class Optimizer:
+    def __init__(self, net, constraints, set_params, solver, obj_fun,
+                 tstop, scale_factor=1., smooth_window_len=None):
+        if net.external_drives:
+            raise ValueError("The current Network instance has external " +
+                             "drives, provide a Network object with no " +
+                             "drives.")
+        self.net = net
```

Comment:
is this the optimized network or the original network? In sklearn terminology, you'd add an underscore at the end of the name if it's optimized network